packages/margin_trading/sources/margin_pool.move

```
@@ -276,29 +234,23 @@ public fun supply<Asset>(
     });
 }
 
-/// Allows withdrawal from the margin pool. Returns the withdrawn coin and the new user supply amount.
+/// Allows withdrawal from the margin pool. Returns the withdrawn coin.
 public fun withdraw<Asset>(
     self: &mut MarginPool<Asset>,
     registry: &MarginRegistry,
-    amount: Option<u64>,
+    percentage: u64,
     clock: &Clock,
     ctx: &mut TxContext,
 ): Coin<Asset> {
     registry.load_inner();
-    self.update_state(clock);
-
     let supplier = ctx.sender();
+    let percentage = percentage.min(constants::float_scaling());
```

Should error out if percentage is too big, instead of defaulting to 1